### PR TITLE
Desktop. Publish SplitPane component, refactor components structure

### PR DESCRIPTION
**Pull request by igordmn**
- publish SplitPane (https://github.com/JetBrains/compose-jb/issues/334)

- refactor components structure (desktop -> demo, common -> library)

- convert line endings in SplitPane to "\n"

### CODE REVIEW

**Comment by igordmn** [comment]
Example of published components: 0.0.0-newci-316

File: components/buildSrc/src/main/kotlin/GradleUtils.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import org.gradle.api.Project
+import org.gradle.api.publish.maven.MavenPublication
+import java.lang.UnsupportedOperationException
+
+inline fun <reified T> Project.configureIfExists(fn: T.() -> Unit) {
+    extensions.findByType(T::class.java)?.fn()
+}
+
+var MavenPublication.mppArtifactId: String
+    get() = throw UnsupportedOperationException()
```
Comment (olonho) [inline components/buildSrc/src/main/kotlin/GradleUtils.kt:15]: Why so?

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposePlugin.kt [hunk 2]
```diff
@@ -149,6 +151,10 @@ class ComposePlugin : Plugin<Project> {
         }
     }
 
+    object DesktopComponentsDependencies {
+        val splitPane = composeDependency("org.jetbrains.compose.components:components-splitpane")
```
Comment (olonho) [inline gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposePlugin.kt:155]: Cool!

**Review by olonho — APPROVED** [review]
Great change, thanks!

File: components/buildSrc/src/main/kotlin/GradleUtils.kt (hunk 1 above)
Comment (igordmn) [reply components/buildSrc/src/main/kotlin/GradleUtils.kt:15]: We don't use this, and implementation would not be trivial

**Pull request merged**